packages/mantle/src/components/combobox/combobox.tsx

```diff
@@ -0,0 +1,191 @@
+	unmountOnHide = true,
+	...props
+}: Primitive.ComboboxPopoverProps) => {
+	return (
+		<Primitive.ComboboxPopover
+			className={cx(
+				"border-popover relative z-50 max-h-96 min-w-[8rem] overflow-hidden rounded-md border shadow-md",
+				// "data-side-bottom:slide-in-from-top-2 data-side-left:slide-in-from-right-2 data-side-right:slide-in-from-left-2 data-side-top:slide-in-from-bottom-2 data-state-closed:animate-out data-state-closed:fade-out-0 data-state-closed:zoom-out-95 data-state-open:animate-in data-state-open:fade-in-0 data-state-open:zoom-in-95",
+				"p-1",
+				"mt-1",
+				"bg-popover",
+				// position === "popper" &&
+				// 	"data-side-bottom:translate-y-2 data-side-left:-translate-x-2 data-side-right:translate-x-2 data-side-top:-translate-y-2 max-h-[var(--radix-select-content-available-height)]",
+				// width === "trigger" && "w-[var(--radix-select-trigger-width)]",
+				className,
+			)}
+			sameWidth={sameWidth}
+			unmountOnHide={unmountOnHide}
+			{...props}
+		>
+			{children}
+		</Primitive.ComboboxPopover>
+	);
+};
+
+const ComboboxItem = ({
+	children,
+	className,
+	focusOnHover = true,
+	...props
+}: Primitive.ComboboxItemProps) => {
+	return (
+		<Primitive.ComboboxItem
+			className={cx(
+				"cursor-pointer rounded px-2 py-1.5 text-sm",
+				"data-active-item:bg-popover-hover",
+				className,
+			)}
+			focusOnHover={focusOnHover}
+			{...props}
+		>
+			{children}
+		</Primitive.ComboboxItem>
+	);
+};
+
+const ComboboxGroup = ({
+	children,
+	className,
+	...props
+}: Primitive.ComboboxGroupProps) => {
+	return (
+		<Primitive.ComboboxGroup className={cx("", className)} {...props}>
+			{children}
+		</Primitive.ComboboxGroup>
+	);
+};
+
+const ComboboxGroupLabel = ({
+	children,
+	className,
+	...props
+}: Primitive.ComboboxGroupLabelProps) => {
+	return (
+		<Primitive.ComboboxGroupLabel
+			className={cx("text-muted px-2 py-1 text-xs font-medium", className)}
+			{...props}
+		>
+			{children}
+		</Primitive.ComboboxGroupLabel>
+	);
+};
+
+type Props = Omit<ComponentProps<"span">, "children"> & {
+	value: string;
+};
+
+function ComboboxHighlightMatch({
+	className: _className,
+	value,
+	...props
+}: Props) {
+	const combobox = Primitive.useComboboxContext();
+	const query = Primitive.useStoreState(combobox, "value");
+	const className = cx("text-strong font-normal", _className);
+
+	if (!query) {
+		return (
+			<span className={className} {...props}>
+				{value}
+			</span>
+		);
+	}
+
+	const lowerText = value.toLowerCase();
+	const lowerQuery = query.toLowerCase();
+
+	if (!lowerText.startsWith(lowerQuery)) {
+		// No highlight if query is not at the start of the value
+		return (
+			<span className={className} {...props}>
+				{value}
+			</span>
+		);
+	}
+
+	const prefix = value.substring(0, query.length);
+	const suffix = value.substring(query.length);
+
+	return (
+		<span className={className} {...props}>
+			<span data-highlighted className="font-bold">
+				{prefix}
+			</span>
```

Comment:
Should we use [mark](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/mark)?

Reply:
maybe! maybe not!

> Don't use <mark> for syntax highlighting purposes; instead, use the [<span>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/span) element with appropriate CSS applied to it.

Reply:
This is above that though (which I thought applies here, _but now I have no idea_ 🤷‍♀️ ):
> indicates a portion of the document's content which is likely to be relevant to the user's current activity. This might be used, for example, to indicate the words that matched a search operation.